homeassistant/components/alexa/entities.py

```diff
@@ -799,8 +798,11 @@ def _check_requirements(self):
             )
             return False
 
-        url = urlparse(network.async_get_external_url(self.hass))
-        if url.scheme != "https":
+        try:
+            network.async_get_url(
+                self.hass, allow_internal=False, prefer_cloud=True, require_ssl=True
```

Comment:
@ochlocracy Quick question, does Alexa require the external URL to be port 443? Or is any other port allowed as well?

Reply:
For the `CameraStreamCapability` to work on official Amazon echo Show devices, the URL returned by stream must be on port 443 using HTTPS protocol.

- The Amazon FireTV Stick doesn't seem to care what port, but the protocol must be HTTPS.
- Any 3rd party Alexa Voice Service devices can choose to handle it however they want. 

Note: This doesn't have to be an "external" URL, the connection is from the echo show device to HA on the users LAN. But it does require HTTPS on port 443.

Reply:
```
require_ssl=True, 
require_standard_port=True, 
allow_internal=False, 
allow_external=True, 
allow_cloud=True, 
allow_ip=False, 
prefer_external=True, 
prefer_cloud=True/False?
```
That my recommendation based on your flow chart, and what the `CameraStreamCapability` requires. 

Although internal URLs can work, it's unlikely that a user has an internal URL with SSL from a trusted authority which echo devices require, it's more likely they have a self-sign SSL. Therefore the external is probably preferred, regarding preference of the cloud URL, I'm unsure.

Reply:
Thanks! That is helpful!

For the cloud, `get_external_url` (which was used before) preferred the cloud over the old `base_url` setting.

If it is really needed? If one has an external URL configured that matches all those requirements, should it really use the cloud?

Reply:
No... at that point I think the only issue would be hairpin NAT issues.